### PR TITLE
Migrate AbstractTaskMapFragment to Jetpack Compose

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Towards https://github.com/google/ground-android/issues/1795

### Description
Migrates `AbstractTaskMapFragment` from legacy XML view binding (`map_task_frag.xml`) to pure Jetpack Compose. 
The map overlay UI is decomposed into a screen-level composable (`TaskMapScreen`) and an extracted leaf component (`LocationInfoCard`) with dedicated unit and integration tests.
### Changes
- **Compose UI**:
  - Add `TaskMapScreen` to host center crosshairs, map type button, recenter button, location lock button, and location info card.
  - Extract `LocationInfoCard` and `@Immutable` data class `LocationInfo` into `tasks/map/components/`.
  - Apply Material 3 theme tokens and Compose API guideline conventions (parameter ordering, weights).
- **Fragment**:
  - Update `AbstractTaskMapFragment` to inflate `BasemapLayoutBinding` and set Compose content with `DisposeOnViewTreeLifecycleDestroyed`.
  - Stream location updates in `onViewCreated` using `viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED)`.
  - Guard against map drag race condition by only updating `_locationInfo` from GPS stream when `locationLock` is active.
  - Hoist `DecimalFormat` to a companion object constant and reset view state in `onDestroyView`.
- **Cleanup**:
  - Remove obsolete `map_task_frag.xml` layout.
  - Remove dead `assertInfoCardShown` / `assertInfoCardHidden` methods from `TaskFragmentRunner`.
- **Tests**:
  - Add `LocationInfoCardTest` testing coordinate display, good/bad accuracy thresholds, unknown accuracy, and null safety.
  - Add `TaskMapScreenTest` verifying center marker, map buttons, recenter toggling, and info card integration.
### Verification
- `./gradlew ktfmtCheck` passed
- `./gradlew detekt` passed (0 issues)
- `./gradlew testLocalDebugUnitTest jvmTest testAndroidHostTest` passed (all tests green)

Before

[Screen_recording_20260909_183305.webm](https://github.com/user-attachments/assets/dfce3875-1992-4a40-a6ff-e9a9a62f4f6b)

After

[Screen_recording_20260909_183035.webm](https://github.com/user-attachments/assets/e084d1c8-c0b0-4887-9456-1f6aa1c67b84)

@andreia-ferreira  PTAL?

### CODE REVIEW

File: app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/map/components/LocationInfoCard.kt [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+/** A floating card displaying current/map coordinates and GPS accuracy. */
+@Composable
+fun LocationInfoCard(locationInfo: LocationInfo, modifier: Modifier = Modifier) {
+  val hasAccuracy = !locationInfo.accuracyText.isNullOrBlank()
+
+  Card(
+    modifier = modifier.testTag(LOCATION_INFO_CARD_TEST_TAG),
+    shape = RoundedCornerShape(8.dp),
+    colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
+  ) {
```
Comment (andreia-ferreira) [inline app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/map/components/LocationInfoCard.kt:66]: the new card is slightly different than the current design. I believe it's missing a border here. Also the locationText color is now a bit darker, perhaps it should also use `onSurfaceVariant`

Old
Image
New
Image

File: app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/AbstractTaskMapFragment.kt [hunk 2]
```diff
@@ -88,64 +86,59 @@ abstract class AbstractTaskMapFragment<TVM : AbstractTaskViewModel> :
...
 
-        if (shouldShowRecenter)
-          RecenterButton(
-            modifier = Modifier.padding(start = 20.dp),
-            onClick = { viewModel.onLocationLockClick() },
-          )
+    viewLifecycleOwner.lifecycleScope.launch {
+      viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+        getMapViewModel().location.collect {
+          if (getMapViewModel().locationLock.value.getOrDefault(false)) {
```
Comment (andreia-ferreira) [inline app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/AbstractTaskMapFragment.kt]: it's safer to  `combine` the `location` and `locationLock` flows so that each emission always carries the latest value of both and race conditions are avoided

File: app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/map/components/LocationInfoCard.kt (hunk 1 above)
Comment (shobhitagarwal1612) [reply app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/map/components/LocationInfoCard.kt:66]: Done

File: app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/AbstractTaskMapFragment.kt (hunk 2 above)
Comment (shobhitagarwal1612) [reply app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/AbstractTaskMapFragment.kt]: Done

**Review by andreia-ferreira — APPROVED** [review]
🚀
